Source/ThirdParty/GltfPipeline/processPbrMetallicRoughness.js

```diff
@@ -0,0 +1,770 @@
+            if (material.hasOwnProperty('pbrMetallicRoughness')) {
+                hasPbrMetallicRoughness = true;
+            }
+        });
+
+        if (hasPbrMetallicRoughness) {
+            if (!defined(gltf.programs)) {
+                gltf.programs = [];
+            }
+            if (!defined(gltf.shaders)) {
+                gltf.shaders = [];
+            }
+            if (!defined(gltf.techniques)) {
+                gltf.techniques = [];
+            }
+
+            // Pre-processing to assign skinning info and address incompatibilities
+            splitIncompatibleSkins(gltf);
+
+            if (!defined(gltf.techniques)) {
+                gltf.techniques = [];
+            }
+            var materials = [];
+            ForEach.material(gltf, function(material) {
+                if (material.hasOwnProperty('pbrMetallicRoughness')) {
+                    var pbrMetallicRoughness = material.pbrMetallicRoughness;
+                    var technique = generateTechnique(gltf, material, options.optimizeForCesium);
+
+                    var values = pbrMetallicRoughness;
+                    for (var valueName in values) {
+                        if (values.hasOwnProperty(valueName)) {
+                            var value = values[valueName];
+                            values[valueName] = value;
+                        }
+                    }
+
+                    var newMaterial = {values : values};
+                    newMaterial.technique = technique;
+                    materials.push(newMaterial);
+                }
+            });
+            gltf.materials = materials;
+
+            // If any primitives have semantics that aren't declared in the generated
+            // shaders, we want to preserve them.
+            ensureSemanticExistence(gltf);
+        }
+
+        return gltf;
+    }
+
+    function generateTechnique(gltf, material, optimizeForCesium) {
+        var techniques = gltf.techniques;
+        var shaders = gltf.shaders;
+        var programs = gltf.programs;
+
+        var parameterValues = material.pbrMetallicRoughness;
+        for (var additional in material) {
+            if (material.hasOwnProperty(additional) && ((additional.indexOf('Texture') >= 0) || additional.indexOf('Factor') >= 0) || additional === 'doubleSided') {
+                parameterValues[additional] = material[additional];
+            }
+        }
+
+        var vertexShader = 'precision highp float;\n';
+        var fragmentShader = 'precision highp float;\n';
+
+        var skin;
+        if (defined(gltf.skins)) {
+            skin = gltf.skins[0];
+        }
+        var joints = (defined(skin)) ? skin.joints : [];
+        var jointCount = joints.length;
+        var hasSkinning = jointCount > 0;
+        var skinningInfo = {};
+        if (hasSkinning) {
+            skinningInfo = material.extras._pipeline.skinning;
+        }
+
+        var hasNormals = true;
```

Comment:
As talked about offline, this should be checked instead of assumed to be `true`.